swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self,inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self,id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        return False
+  
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+        self.inventory[0] = their_first_item
+        other_vendor.inventory[0] = my_first_item
+        return True
+
+    def get_by_category(self,category):
+        list_by_category = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                list_by_category.append(item)
+        return list_by_category
+    
+    def get_best_by_category(self,category):
+        best_item = None
+        for item in self.inventory:
+            if item.get_category() == category:
+                if best_item is None or item.condition > best_item.condition:
+                    best_item = item
+        return best_item
```

Comment:
This is a little bit of a nitpick, but in terms of readability, it can be helpful to add in an empty line between different parts of your function! One really common pattern you will see is separating out the setup, the functionality and the return. This might look something like this:

```suggestion
        #setup
        best_item = None 
        
        #functionality
        for item in self.inventory:
            if item.get_category() == category:
                if best_item is None or item.condition > best_item.condition:
                    best_item = item
         
        #return                      
        return best_item
```

We end up with some separation which can really help readability! Feel free to apply this pattern to other functions you have as well!